src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -94,6 +95,53 @@ public Mono<Tuple2<Mono<TransactionActivatedEvent>, String>> handle(
                                                 getPaymentRequestInfoFromCache(new RptId(paymentNotice.getRptId()))
                                         )
                                 )
+                        ).flatMap(
+                                paymentRequest -> {
+                                    final PaymentNoticeInfoDto paymentNotice = paymentRequest.getT1();
+                                    final Optional<PaymentRequestInfo> maybePaymentRequestInfo = paymentRequest
+                                            .getT2();
+                                    return Mono.just(
+                                            Tuples.of(
+                                                    paymentNotice,
+                                                    Optional.of(
```

Comment:
Since paymentRequestInfo is always initialized even in case of cache miss can we remove the Optional from the returned Tuple here and return directly the payment request info obtained from the Optional.orElseGet chain?

Reply:
correct :-)

Reply:
Solved with d6101c8